stl_to_obj/stl_to_tscn.py

```diff
@@ -0,0 +1,209 @@
+                self.sub_resource += f'[sub_resource type="SpatialMaterial" id={self.sub_resource_id}]\n'
+
+                for jpg, value in t_info.jpg_dict.items():
+                    if jpg in ["albedo", "roughness", "metallic"]:
+                        self.sub_resource += f"{jpg}_texture = ExtResource( {self.ext_resource_id} )\n"
+                    elif jpg == "normal":
+                        self.sub_resource += Textures.NORMAL_OPTIONS + f"normal_texture = ExtResource( {self.ext_resource_id} )\n"
+                    elif jpg == "depth":
+                        self.sub_resource += Textures.DEPTH_OPTIONS + f"depth_texture = ExtResource( {self.ext_resource_id} )\n"
+                    # add .jpg as ext_resource
+                    self.ext_resource += f'[ext_resource path="res://{Textures.TEXTURE_FOLDER}/{t_info.folder}/{value}" type="Texture" id={self.ext_resource_id}]\n'
+                    self.ext_resource_id += 1
+
+                # add uv1_scale if texture has it
+                if t_info.uv1_scale:
+                    self.sub_resource += f"uv1_scale = {t_info.uv1_scale}\n"
+
+            # handle textures which don't need .jpgs (only need sub_resource)
+            elif texture in Textures.other_textures:
+                self.sub_resource += f'[sub_resource type="SpatialMaterial" id={self.sub_resource_id}]\n{Textures.other_textures[texture]}'
+
+            self.sub_resource += "\n"
+                
+            # add the sub resource's index to texture_index so we can use it later 
+            self.texture_index[texture] = self.sub_resource_id
+            self.sub_resource_id += 1 
+
+    def add_obj_file(self, stl_file_extless, uv_map, mesh_compression, texture):
+        # reset bpy
+        bpy.ops.wm.read_factory_settings(use_empty=True)
+        # import .stl into bpy
+        bpy.ops.import_mesh.stl(filepath = f"{self.input_folder}/{stl_file_extless}.stl")
+        # Get all objects in selection
+        selection = bpy.context.selected_objects
+        # Deselect all objects
+        bpy.ops.object.select_all(action='DESELECT')
+
+        for obj in selection:
+            # Select each object
+            obj.select_set(True)
+            # Make it active
+            bpy.context.view_layer.objects.active = obj
+            # Toggle into Edit Mode
+            bpy.ops.object.mode_set(mode='EDIT')
+            # Select the geometry
+            bpy.ops.mesh.select_all(action='SELECT')
+
+            # Remove unnecessary faces 
+            if mesh_compression == "limited_dissolve": 
+                bpy.ops.mesh.dissolve_limited()
+            elif mesh_compression == "tris_to_quads":
+                bpy.ops.mesh.tris_convert_to_quads() 
+
+            # Call project operator to generate uv maps
+            try:
+                UV_MAPS[uv_map]()
+            except KeyError as e:
+                print(f"no uv map generated for {stl_file_extless} due to invalid uv_map")
```

Comment:
```suggestion
            except KeyError as e:
                print(f"no uv map generated for {stl_file_extless} due to invalid uv_map")
                raise e
```
If you're going to store the `KeyError`, you may as well use it. Otherwise, you can avoid storing it by removing `as e` from the `except` line:
```suggestion
            except KeyError:
                e = KeyError(f"no uv map generated for {stl_file_extless} due to invalid uv_map")
                raise e
```
From my understanding, you generally want to avoid print statements when raising an exception, and instead want to raise the exception itself to make it clearer to the user what went wrong.